### PR TITLE
feat: Versionable `Message`

### PR DESCRIPTION
**Pull request by bvrooman**
Related issues:
- https://github.com/FuelLabs/fuel-core/issues/1552

This PR converts the `Message` struct to an enum that can house multiple variants for versioning. Version variants contain an instance of a versioned struct, e.g., `MessageV1`. The `Message` enum exposes getters to the underlying data and abstracts the version. Setters are exposed only in test environments to allow tests to set up `Messages` in specific ways for the test.

### CODE REVIEW

File: crates/services/relayer/src/ports/tests.rs [hunk 1]
```diff
@@ -22,13 +22,16 @@ fn test_insert_messages() {
         .returning(|_| Ok(Some(std::borrow::Cow::Owned(9u64.into()))));
     let mut db = db.into_transactional();
 
-    let m = Message {
-        amount: 10,
-        da_height: 12u64.into(),
-        ..Default::default()
-    };
+    // let m = Message {
```
Comment (MitchTurner) [inline crates/services/relayer/src/ports/tests.rs:25]: Do you need to remove this comment?

**Review by MitchTurner — DISMISSED** [review]
LGTM. I think it's just that comment that needs to be removed.

**Review by MitchTurner — DISMISSED** [review]

File: crates/types/src/entities/message.rs [hunk 2]
```diff
@@ -24,10 +24,26 @@ use crate::{
     },
 };
 
+/// Versioned Message
```
Comment (xgreenx) [inline crates/types/src/entities/message.rs:27]:
```suggestion
/// Message sent from DA layer to fuel by relayer bridge.
```

File: crates/types/src/entities/message.rs [hunk 3]
```diff
@@ -24,10 +24,26 @@ use crate::{
...
+}
+
+#[cfg(any(test, feature = "test-helpers"))]
+impl Default for Message {
+    fn default() -> Self {
+        Self::V1(Default::default())
+    }
+}
+
 /// Message send from Da layer to fuel by bridge
```
Comment (xgreenx) [inline crates/types/src/entities/message.rs:43]:
```suggestion
/// The V1 version of the message from the DA layer.
```

**Review by xgreenx — APPROVED** [review]

**Review by xgreenx — APPROVED** [review]

**Pull request merged**